### DISCUSSION TITLE
Document library list creation returning `HRESULT: 0x8107140F` exception

### DISCUSSION THREAD

**Discussion by Amwins-Vault-Boy** [pnp/pnpcore#1450] [Q&A] [answered]
We are having an odd issue when creating document library lists. We are getting the following exception when creating it via ```context.Web.Lists.AddAsync(listName, ListTemplateType.DocumentLibrary);```

```
HttpResponseCode: 500
Code: Microsoft.SharePoint.SPException
Message: Exception from HRESULT: 0x8107140F
ClientRequestId: 388422a1-006e-5000-25f8-6703fc5b533a
SPClientServiceRequestDuration: 204
X-SharePointHealthScore: 0
X-SP-SERVERSTATE: ReadOnly=0
```

The weird part is that it actually creates the list successfully.

We are trying to work with Microsoft on it, but they wanted a Correlation Id. The problem is `PnpCorrelationId` is an empty GUID.

Any ideas on (1) what might be causing this exception, and (2) why the correlation id is empty?

Thanks!

**Comment by jansenbe** [comment]
@Amwins-Vault-Boy : no idea why this is happening, isn't `ClientRequestId` not the correlation id? The PnP correlation id is not useful for support

**Reply by Amwins-Vault-Boy** [reply, marked as answer]
Microsoft insisted the request id was not the correlation id, but I agree with you. I think they are wrong, but I think we resolved the issue now. Thanks for taking the time to respond!
